container/container.go

```diff
@@ -48,3 +49,25 @@ func ContainerStatus(namespace []string) (status StatusType, err error) {
 	}
 	return
 }
+
+// WaitForContainerStatus wait for the container to have the provided status until it reach the timeout
+func WaitForContainerStatus(namespace []string, status StatusType, timeout time.Duration) (err error) {
+	start := time.Now()
+	for {
+		currentStatus, err := ContainerStatus(namespace)
+		if err != nil {
+			return err
```

Comment:
I would use some `break` here in order to make sure that we always exit the for loop, it's a bit easier to test

Reply:
Because of the new context of the `for` and the creation of `currentStatus`, `err` is also created. So if I use `break`, I'm pretty sure the err will not be assign to the return `err`.
If I write, the following code, go says `err is shadowed during return` on the `return`.
```go
for {
    currentStatus, err := ContainerStatus(namespace)
    if err != nil {
	return
```

So there are 2 solutions.
The first one is to use `return err` as it's already the case.
The second is to created `currentStatus` and then assign the value:
```go
var currentStatus StatusType
currentStatus, err = ContainerStatus(namespace)
if err != nil {
	break
}
```

@antho1404 What solution do you think it's the best?

Reply:
I personally use the second solution all the time, we should definitely avoid shadowed variable it's the kind of things that can create bugs easily.